mlperf_common/callbacks/logging.py

```diff
@@ -0,0 +1,349 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+try:
+    import lightning.pytorch as pl
+    from lightning.pytorch.loggers import Logger
+    from lightning.pytorch.utilities import rank_zero_only
+except ImportError:
+    import pytorch_lightning as pl
+    from pytorch_lightning.loggers import Logger
+    from pytorch_lightning.utilities import rank_zero_only
+
+import os
+import time
+from typing import Any, Callable, Literal
+
+from mlperf_common.frameworks.pyt import PyTCommunicationHandler
+from mlperf_common.logging import MLLoggerWrapper
+
+mllogger = MLLoggerWrapper(PyTCommunicationHandler())
+
+
+class DeltaTimer:
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self.start_time = time.perf_counter()
+        return self.start_time
+
+    def get_delta(self):
+        prev_time = self.start_time
+        return self.reset() - prev_time
+
+
+class LoggingCallback(pl.Callback):
+    def __init__(self, force_success: bool | None = None):
```

Comment:
why do we even want to handle `force_success` as a parameter? It should be govern by an envvar everywhere, right?

Reply:
I am keeping behaviour that if not specified, the force_success will be readed from envvar
But for maintability reasons, I would like to avoid pure envvar based parameters